RedHat-UX/red-hat-design-system
# docs(rh-dialog): update docs

Closes #927

## What I did

1. Updated the documentation for the dialog element according to the specs outlined in #927.


## Testing Instructions

1. Review latest docs at https://deploy-preview-1001--red-hat-design-system.netlify.app/elements/dialog and verify they match the design specs as well as the docs cleanup guide https://github.com/RedHat-UX/red-hat-design-system/wiki/Docs-Cleanup-Q2-2023-Checklist#code-page.
